logger panics on key errors

looks like this was introduced with #3660. After running the logger for some time, it will panic with the following error:

```
><> docker logs deis-logger-git-666c7fc
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x10 pc=0x4017b0]

goroutine 1 [running]:
main.main()
	/Users/bacongobbler/go/src/github.com/deis/deis/logger/main.go:70 +0x7e0

goroutine 5 [syscall, 5 minutes]:
os/signal.loop()
	/usr/local/go/src/os/signal/signal_unix.go:21 +0x1f
created by os/signal.init·1
	/usr/local/go/src/os/signal/signal_unix.go:27 +0x35

goroutine 9 [runnable]:
net/http.(*persistConn).writeLoop(0xc20805c160)
	/usr/local/go/src/net/http/transport.go:945 +0x41d
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:661 +0xcbc

goroutine 11 [select, 5 minutes]:
github.com/deis/deis/logger/syslogd.Listen(0xc2080582a0, 0xc208058300, 0xc2080581e0, 0xc2080331d0, 0xd)
	/Users/bacongobbler/go/src/github.com/deis/deis/logger/syslogd/syslogd.go:123 +0x8a1
created by main.main
	/Users/bacongobbler/go/src/github.com/deis/deis/logger/main.go:62 +0x65d

goroutine 12 [runnable]:
net/http.(*persistConn).roundTrip(0xc20805c0b0, 0xc208253a70, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/http/transport.go:1082 +0x7ad
net/http.(*Transport).RoundTrip(0xc208072090, 0xc208242b60, 0xc208256b40, 0x0, 0x0)
	/usr/local/go/src/net/http/transport.go:235 +0x558
net/http.send(0xc208242b60, 0x7f62d7201d80, 0xc208072090, 0x32, 0x0, 0x0)
	/usr/local/go/src/net/http/client.go:219 +0x4fc
net/http.(*Client).send(0xc20803cc30, 0xc208242b60, 0x32, 0x0, 0x0)
	/usr/local/go/src/net/http/client.go:142 +0x15b
net/http.(*Client).doFollowingRedirects(0xc20803cc30, 0xc208242b60, 0x7b2218, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/http/client.go:367 +0xb25
net/http.(*Client).Do(0xc20803cc30, 0xc208242b60, 0xc, 0x0, 0x0)
	/usr/local/go/src/net/http/client.go:177 +0x192
github.com/coreos/go-etcd/etcd.(*Client).SendRequest(0xc208068000, 0xc20824d2c0, 0x0, 0x0, 0x0)
	/Users/bacongobbler/go/src/github.com/deis/deis/Godeps/_workspace/src/github.com/coreos/go-etcd/etcd/requests.go:233 +0x10db
github.com/coreos/go-etcd/etcd.(*Client).put(0xc208068000, 0xc208253910, 0xf, 0xc2080331e0, 0x5, 0x0, 0x0, 0xc208253910, 0x0, 0x0)
	/Users/bacongobbler/go/src/github.com/deis/deis/Godeps/_workspace/src/github.com/coreos/go-etcd/etcd/requests.go:85 +0x4be
github.com/coreos/go-etcd/etcd.(*Client).RawSet(0xc208068000, 0xc208253910, 0xf, 0xc2080331e0, 0x5, 0x0, 0x71ac50, 0x0, 0x0)
	/Users/bacongobbler/go/src/github.com/deis/deis/Godeps/_workspace/src/github.com/coreos/go-etcd/etcd/set_update_create.go:108 +0x7e
github.com/coreos/go-etcd/etcd.(*Client).Set(0xc208068000, 0xc208253910, 0xf, 0xc2080331e0, 0x5, 0x0, 0x5, 0x0, 0x0)
	/Users/bacongobbler/go/src/github.com/deis/deis/Godeps/_workspace/src/github.com/coreos/go-etcd/etcd/set_update_create.go:7 +0x78
main.setEtcd(0xc208068000, 0xc208253910, 0xf, 0xc2080331e0, 0x5, 0x0)
	/Users/bacongobbler/go/src/github.com/deis/deis/logger/main.go:95 +0x74
main.publishService(0xc2080582a0, 0xc208068000, 0x7fffc2c46f4e, 0xe, 0x72c830, 0xa, 0xc2080331e0, 0x5, 0x0)
	/Users/bacongobbler/go/src/github.com/deis/deis/logger/main.go:87 +0x19d
created by main.main
	/Users/bacongobbler/go/src/github.com/deis/deis/logger/main.go:64 +0x74c

goroutine 14 [chan receive, 5 minutes]:
github.com/deis/deis/logger/syslogd.(*handler).mainLoop(0xc20801eb20)
	/Users/bacongobbler/go/src/github.com/deis/deis/logger/syslogd/syslogd.go:88 +0x8f
created by github.com/deis/deis/logger/syslogd.newHandler
	/Users/bacongobbler/go/src/github.com/deis/deis/logger/syslogd/syslogd.go:36 +0x187

goroutine 15 [IO wait, 4 minutes]:
net.(*pollDesc).Wait(0xc208010ca0, 0x72, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:84 +0x47
net.(*pollDesc).WaitRead(0xc208010ca0, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:89 +0x43
net.(*netFD).readFrom(0xc208010c40, 0xc20809c000, 0x100000, 0x100000, 0x0, 0x0, 0x0, 0x7f62d7201a48, 0xc2080322c0)
	/usr/local/go/src/net/fd_unix.go:269 +0x4a1
net.(*UDPConn).ReadFromUDP(0xc208040138, 0xc20809c000, 0x100000, 0x100000, 0x64ab80, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/udpsock_posix.go:67 +0x124
net.(*UDPConn).ReadFrom(0xc208040138, 0xc20809c000, 0x100000, 0x100000, 0x100000, 0x0, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/udpsock_posix.go:82 +0x12e
github.com/deis/deis/logger/syslog.(*Server).receiver(0xc20800a870, 0x7f62d7203680, 0xc208040138)
	/Users/bacongobbler/go/src/github.com/deis/deis/logger/syslog/server.go:104 +0xaf
created by github.com/deis/deis/logger/syslog.(*Server).Listen
	/Users/bacongobbler/go/src/github.com/deis/deis/logger/syslog/server.go:64 +0x2e3

goroutine 17 [select]:
net/http.(*persistConn).readLoop(0xc20805c0b0)
	/usr/local/go/src/net/http/transport.go:928 +0x9ce
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:660 +0xc9f

goroutine 18 [select]:
net/http.(*persistConn).writeLoop(0xc20805c0b0)
	/usr/local/go/src/net/http/transport.go:945 +0x41d
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:661 +0xcbc
```

My guess is that the [node value incoming](https://github.com/deis/deis/blob/master/logger/main.go#L69-L70) from client watch is nil, but we're still trying to extract the value from a nil node.

--- a/logger/main.go
+++ b/logger/main.go
@@ -66,8 +66,10 @@ func main() {
 
 	for {
 		select {
-		case er := <-drainRespChan:
-			drainChan <- er.Node.Value
+		case dr := <-drainRespChan:
+			if dr != nil && dr.Node != nil {
+				drainChan <- dr.Node.Value
+			}
 		case <-signalChan:
 			close(exitChan)
 			stopChan <- true

